src/plugins/auto/auto_schedule.cpp

```diff
@@ -218,7 +310,11 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
                         });
                     }
                 }
-                contextPtr->promise.set_value();
+                try {
+                    contextPtr->promise.set_value();
```

Comment:
something minor you can have a try if want to get rid of the try block, before trying to load for second failure, clear the promise:
contextPtr->promise = {};
and in loading:
contextPtr->future = contextPtr->promise.get_future().share();